File: app_flutter/lib/models/device.dart
class Device {
  final int id;
  final String deviceId;
  final String deviceName;
  final String deviceType;
  final String? location;
  final String status;
  final String? currentState;
  final String? firmware;
  final int? rssi;
  final DateTime registeredAt;
  final DateTime lastSeen;
  
  Device({
    required this.id,
    required this.deviceId,
    required this.deviceName,
    required this.deviceType,
    this.location,
    required this.status,
    this.currentState,
    this.firmware,
    this.rssi,
    required this.registeredAt,
    required this.lastSeen,
  });
  
  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      id: json['id'],
      deviceId: json['deviceId'],
      deviceName: json['deviceName'],
      deviceType: json['deviceType'],
      location: json['location'],
      status: json['status'],
      currentState: json['currentState'],
      firmware: json['firmware'],
      rssi: json['rssi'],
      registeredAt: DateTime.parse(json['registeredAt']),
      lastSeen: DateTime.parse(json['lastSeen']),
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'deviceId': deviceId,
      'deviceName': deviceName,
      'deviceType': deviceType,
      'location': location,
      'status': status,
      'currentState': currentState,
      'firmware': firmware,
      'rssi': rssi,
      'registeredAt': registeredAt.toIso8601String(),
      'lastSeen': lastSeen.toIso8601String(),
    };
  }
}

File: iot_backend/src/main/java/com/iot/backend/model/Device.java
package com.iot.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "devices")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Device {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "device_id", unique = true, nullable = false, length = 100)
    private String deviceId;
    
    @Column(name = "device_name", nullable = false, length = 200)
    private String deviceName;
    
    @Column(name = "device_type", nullable = false, length = 50)
    private String deviceType;
    
    @Column(length = 100)
    private String location;
    
    @Column(length = 50)
    private String status = "offline";
    
    @Column(name = "current_state", length = 50)
    private String currentState;
    
    @Column(length = 100)
    private String firmware;
    
    private Integer rssi;
    
    @Column(name = "registered_at")
    private LocalDateTime registeredAt;
    
    @Column(name = "last_seen")
    private LocalDateTime lastSeen;
    
    @PrePersist
    protected void onCreate() {
        if (registeredAt == null) {
            registeredAt = LocalDateTime.now();
        }
        if (lastSeen == null) {
            lastSeen = LocalDateTime.now();
        }
        if (status == null) {
            status = "offline";
        }
    }
    
    @PreUpdate
    protected void onUpdate() {
        lastSeen = LocalDateTime.now();
    }
}

